src/main/java/uk/gov/hmcts/reform/iacasenotificationsapi/domain/personalisation/appellant/letter/AppellantInternalDecideApplicationLetterPersonalisation.java

```diff
@@ -0,0 +1,113 @@
+    }
+
+    @Override
+    public Set<String> getRecipientsList(final AsylumCase asylumCase) {
+        return getAppellantAddressInCountryOrOoc(asylumCase);
+    }
+
+    @Override
+    public String getReferenceId(Long caseId) {
+        return caseId + "_INTERNAL_DECIDE_APPLICATION_LETTER";
+    }
+
+    @Override
+    public Map<String, String> getPersonalisation(Callback<AsylumCase> callback) {
+        requireNonNull(callback, "callback must not be null");
+
+        AsylumCase asylumCase =
+            callback
+                .getCaseDetails()
+                .getCaseData();
+
+        YesOrNo isAppellantInUK = asylumCase.read(AsylumCaseDefinition.APPELLANT_IN_UK, YesOrNo.class).orElse(YesOrNo.NO);
+
+        List<String> appellantAddress = switch (isAppellantInUK) {
+            case YES -> getAppellantAddressAsList(asylumCase);
+            case NO -> getAppellantAddressAsListOoc(asylumCase);
+        };
+
+        MakeAnApplication application = getMakeAnApplication(asylumCase).orElseThrow(() -> new IllegalStateException("Decide application is missing"));
+        String nextSteps = WhatHappensNextContentUtils.getWhatHappensNextContent(
+            MakeAnApplicationTypes.from(application.getType()).orElseThrow(() -> new IllegalStateException("Invalid MakeAnApplicationType")),
+            true, application.getDecision());
+        if (nextSteps.equals("Unknown")) {
+            throw new IllegalStateException("Invalid MakeAnApplicationType: Couldn't find next steps.");
+        }
+
+        boolean isRefused = application.getDecision().equals("Refused");
+        String decisionStr = isRefused ? "refuse" : "grant";
+        ImmutableMap.Builder<String, String> personalizationBuilder = ImmutableMap
+            .<String, String>builder()
+            .putAll(customerServicesProvider.getCustomerServicesPersonalisation())
+            .put("decisionMaker", application.getDecisionMaker())
+            .put("decision", decisionStr)
+            .put("applicationType", application.getType())
+            .put("decisionReason", application.getDecisionReason())
+            .put("nextSteps", nextSteps)
+            .put("appealReferenceNumber", asylumCase.read(AsylumCaseDefinition.APPEAL_REFERENCE_NUMBER, String.class).orElse(""))
+            .put("homeOfficeReferenceNumber", asylumCase.read(AsylumCaseDefinition.HOME_OFFICE_REFERENCE_NUMBER, String.class).orElse(""))
+            .put("appellantGivenNames", asylumCase.read(AsylumCaseDefinition.APPELLANT_GIVEN_NAMES, String.class).orElse(""))
+            .put("appellantFamilyName", asylumCase.read(AsylumCaseDefinition.APPELLANT_FAMILY_NAME, String.class).orElse(""));
+
+        for (int i = 0; i < appellantAddress.size(); i++) {
+            personalizationBuilder.put("address_line_" + (i + 1), appellantAddress.get(i));
+        }
+        return personalizationBuilder.build();
+    }
+
+    Optional<MakeAnApplication> getMakeAnApplication(AsylumCase asylumCase) {
```

Comment:
Is this method has no private modifier on purpose?

Reply:
Fixed in latest commit.